cognite/client/_api/functions.py

```diff
@@ -0,0 +1,1136 @@
+    file_extension = Path(function_path).suffix
+    if file_extension != ".py":
+        raise TypeError(f"{function_path} is not a valid value for function_path. File extension must be .py.")
+
+    function_path_full = Path(root_path) / Path(
+        function_path
+    )  # This converts function_path to a Windows path if running on Windows
+    if not function_path_full.is_file():
+        raise TypeError(f"No file found at location '{function_path}' in '{root_path}'.")
+
+    sys.path.insert(0, root_path)
+
+    # Necessary to clear the cache if you have previously imported the module (this would have precedence over sys.path)
+    cached_handler_module = sys.modules.get("handler")
+    if cached_handler_module:
+        del sys.modules["handler"]
+
+    module_path = convert_file_path_to_module_path(function_path)
+    handler = importlib.import_module(module_path)
+
+    if "handle" not in handler.__dir__():
+        raise TypeError(f"{function_path} must contain a function named 'handle'.")
+
+    _validate_function_handle(handler.handle)
+    sys.path.remove(root_path)
+
+
+def _validate_function_handle(function_handle: Callable[..., Any]) -> None:
+    if not function_handle.__code__.co_name == "handle":
+        raise TypeError("Function referenced by function_handle must be named handle.")
+    if not set(function_handle.__code__.co_varnames[: function_handle.__code__.co_argcount]).issubset(
+        set(["data", "client", "secrets", "function_call_info"])
+    ):
+        raise TypeError(
+            "Arguments to function referenced by function_handle must be a subset of (data, client, secrets, function_call_info)"
+        )
+
+
+def _assert_exactly_one_of_function_id_and_function_external_id(
+    function_id: Optional[int], function_external_id: Optional[str]
+) -> None:
+    utils._auxiliary.assert_type(function_id, "function_id", [Integral], allow_none=True)
+    utils._auxiliary.assert_type(function_external_id, "function_external_id", [str], allow_none=True)
+    has_function_id = function_id is not None
+    has_function_external_id = function_external_id is not None
+
+    assert (has_function_id or has_function_external_id) and not (
+        has_function_id and has_function_external_id
+    ), "Exactly one of function_id and function_external_id must be specified"
+
+
+def _assert_at_most_one_of_function_id_and_function_external_id(
+    function_id: Optional[int], function_external_id: Optional[str]
+) -> None:
+    utils._auxiliary.assert_type(function_id, "function_id", [Integral], allow_none=True)
+    utils._auxiliary.assert_type(function_external_id, "function_external_id", [str], allow_none=True)
+    has_function_id = function_id is not None
+    has_function_external_id = function_external_id is not None
+    assert not (
+        has_function_id and has_function_external_id
+    ), "Only function_id or function_external_id allowed when listing schedules."
```

Comment:
Can use `IdentifierSequence(id, extid).assert_singleton()`?

Reply:
I think it's a good idea to use `assert IdentifierSequence(id, extid).is_singleton()` inside these methods and keep the custom error message. The error message raised from inside `IdentifierSequence(id, extid).assert_singleton()` is misleading in these contexts:
```
Exactly one of id or external id must be specified"
```
What do you think @erlendvollset and @alchemistake ?

Reply:
Ah I'm ahead of you :D

Reply:
Yesterday i wrapped those errors to work like what we already have in calls and other object with a small helper function.

Reply:
https://github.com/cognitedata/cognite-sdk-python/pull/949/commits/f69f9c22cb47eba964c87809aa866519310d1ea5#diff-264fa2713547ccdb770b2891f2b90fab2fe12ad37afb09de63d4db4904c62e13R66

Reply:
Aha, I didn't notice that. Great! 😄 Nitpick: You could save a line or two by using the method `is_singletong()` directly instead of the try/catch?

Reply:
Okay ill convert to ifelse :+1: